### ISSUE TITLE
[BUG] NoneType model value examples are not supported

### ISSUE THREAD

**Issue by Kylmakalle** [tiangolo/fastapi#2149]
### Description
I'm not able to pass a `None`/ as an example for a model value created with `pydantic.Field`

Looks like parser can't find a difference between  `example=None` as a NoneType variable and "_missing example field_"

```Python
class SomeResponse(BaseModel):
       city: Optional[str] = Field(..., nullable=True, example=None)
```

### Expected
```json
"city": {
"title": "City",
"type": "string",
"nullable": true,
"example": null
}
```

### Occurred 
```json
"city": {
"title": "City",
"type": "string",
"nullable": true
}
```

### Environment

* OS: macOS
* FastAPI Version: 0.61.1
* pydantic Version: 1.6.1

* Python version: 3.7.9

**Comment by ycd** [comment]
I did not understand the problem here, why do you need to pass `example=None` in Field.

Also when I check the schema that Pydantic creates, everything seemed fine to me.

```python

SomeResponse.schema()

Out: {
    "properties":{
        "city":{
            "example":"None",
            "nullable":True,
            "title":"City",
            "type":"string"
        }
    },
    "required":[
        "city"
    ],
    "title":"SomeResponse",
    "type":"object"}
```

Can you reproduce how did you get the output?

**Comment by Kylmakalle** [comment]
> why do you need to pass example=None in Field

That's part of api response and it's intended. 

> I check the schema that Pydantic creates

But it looks **not** fine in `/openapi.json`


> Can you reproduce how did you get the output?

It's provided in issue description
> city: Optional[str] = Field(..., nullable=True, example=None)

I also attach an example there so you can be sure it's even more buggy if I provide example value via `class Config` for schema. Fields with `None` as example are missing...
https://gist.github.com/Kylmakalle/8682416c2155b99a89961d33c0d3988b

![image](https://user-images.githubusercontent.com/24507532/96488266-60719800-1246-11eb-8f6c-0798b024fcba.png)

**Comment by ycd** [comment]
Yes I can confirm this is strange, even more strange thing is when i debug your gist,  `scheme_extra` is an empty dictionary.

![image](https://user-images.githubusercontent.com/62724709/96493840-27d5bc80-124e-11eb-8be6-493a94c88540.png)

**Comment by sommergis** [comment]
Hi all,
I ran into the same issue today. It seems like None values of fields in examples in OpenAPI schema generation are being excluded by FastAPI. The API works as expected, it's only an issue in the OpenAPI schema.

`python

class APIErrorMessage(BaseModel):
    """ Model for basic API response for an unsuccessful request. """
    content: Any = None
    error: str = ""
    detail: Optional[str] = None

    class Config:
        """ Real world example for OpenAPI documentation """
        schema_extra = {
            "example": {
                "content": None,
                "error": "Authentication Error",
                "detail": "key validity expired"
            }
        }`

Maybe this is the location in the code? `exclude_none=True`

https://github.com/tiangolo/fastapi/blob/d2eb4a71eef08e201082d23517d4ddc99ff258b5/fastapi/openapi/utils.py#L377

    return jsonable_encoder(OpenAPI(**output), by_alias=True, exclude_none=True)  # type: ignore


**Environment**

FastAPI Version: 0.62.0

**Comment by abuendia** [comment]
Hi there - is there any update on this issue / ETA for the PR merge? @four43

**Comment by four43** [comment]
My PR is open and waiting for approval, I guess I don't know what more I can do :(

**Comment by abuendia** [comment]
Hi @tiangolo - Could you please take a look at this?

**Comment by latimer1** [comment]
There's related issue in pydantic about this topic: https://github.com/samuelcolvin/pydantic/issues/1270

I was able to get my system outputting the correct nullable by overriding the BaseModel:
`class BaseModel(PydanticBaseModel):
    class Config:

        # Due to a bug in pydantic, Optional fields do not automatically set nullable in the openapi.json spec
        # See:
        #   https://github.com/samuelcolvin/pydantic/issues/1270
        #   https://github.com/tiangolo/fastapi/issues/2149
        
        # Without using the basemodel, client libraries will raise an exception if None is specified
        # Example: 
        #       steadilyapi.exceptions.ApiTypeError: Invalid type for variable 'company_name'. 
        #       Required value type is str and passed type was NoneType 
        #       at ['received_data']['primary_insured']['company_name']

        @staticmethod
        def schema_extra(schema, model):
            for prop, value in schema.get('properties', {}).items():
                # retrieve right field from alias or name
                field = [x for x in model.__fields__.values() if x.alias == prop][0]
                if field.allow_none:
                    if '$ref' in value:
                        if issubclass(field.type_, PydanticBaseModel):
                            # add 'title' in schema to have the exact same behaviour as the rest
                            value['title'] = field.type_.__config__.title or field.type_.__name__
 
                    value['nullable'] = True`
